drivers/flash/spi_nor.c

```diff
@@ -540,6 +540,45 @@ static int spi_nor_wrsr(const struct device *dev,
 	return ret;
 }
 
+#define MX25R_HIGH_PERF_MODE BIT(1)
+
+#ifdef CONFIG_SPI_NOR_MX_HIGH_PERF_MODE
+static int mx_specific_enable_high_perf_mode(const struct device *dev, uint8_t jedec_id[])
+{
+	int sr, ret;
+	uint8_t data[3];
+
+	/* Only supported on Macronix MX25R Ultra Low Power series. */
+	if (jedec_id[0] != 0xc2 || jedec_id[1] != 0x28) {
+		return -ENOTSUP;
+	}
+
+	acquire_device(dev);
+
+	/* The configuration register bytes on the Macronix MX25R devices are set using
+	 * the Write Status Register command. First read out the current status register.
+	 */
+	sr = spi_nor_rdsr(dev);
+	if (sr < 0) {
+		return sr;
+	}
+
+	data[0] = sr;
+	data[1] = 0x0;			/* Configuration register 1 */
```

Comment:
I think you should preserve the current value of this configuration register, not just always write it with zero.